File: test_tts.py
import time
from tts import VoiceManager, check_under_voltage

def run_test():
    # 1. 인스턴스 생성
    print("=== TTS 모듈 테스트 시작 ===")
    voice = VoiceManager()

    # 2. 기본 음성 출력 테스트
    print("테스트 1: 기본 출력")
    voice.speak("시스템 테스트를 시작합니다.", priority=2)
    time.sleep(2) # 음성이 나올 시간을 줍니다.

    # 3. 우선순위 큐 테스트 (중요!)
    # 일반 정보들을 먼저 넣고, 바로 뒤에 긴급 메시지를 넣었을 때 
    # 긴급 메시지가 먼저(또는 중간에) 새치기해서 나오는지 확인합니다.
    print("테스트 2: 우선순위 큐 (일반 정보 여러 개 입력 후 긴급 입력)")
    voice.speak("첫 번째 사물입니다.", priority=3)
    voice.speak("두 번째 사물입니다.", priority=3)
    voice.speak("세 번째 사물입니다.", priority=3)
    
    # 순서상 뒤에 넣었지만, priority가 1이므로 큐에서 가장 먼저 뽑힙니다.
    voice.speak("긴급 상황 발생! 전압을 확인하세요.", priority=1)
    
    # 4. 중복 방지(Debounce) 테스트
    print("테스트 3: 중복 방지 (똑같은 말을 짧은 시간에 여러 번 요청)")
    # '사람이 있습니다'를 3번 연속 요청해도 한 번만 나와야 정상입니다.
    for _ in range(3):
        voice.speak("사람이 있습니다.", priority=3, debounce_time=3)
        time.sleep(0.1)

    # 모든 음성이 끝날 때까지 잠시 대기
    print("\n모든 음성 요청 완료. 5초 후 테스트를 종료합니다.")
    time.sleep(5)

if __name__ == "__main__":
    run_test()

File: test_tts.c
#include <stdio.h>
#include <unistd.h>
#include "tts.h" // 음성 매니저 헤더 불러오기

int main() {
    // 1. 시스템 기동 시 최초 1회 초기화
    init_voice_manager();
    
    printf("🚀 [Pineapple Vision] 시스템 루프 구동\n");
    
    // 2. 일반 안내 방송 요청 (우선순위 3, 디바운스 2초)
    speak("카메라 스트리밍을 시작합니다.", 3, 2.0);
    sleep(2);
    
    speak("전방에 의자가 있습니다.", 3, 2.0);
    
    // 동일 문장 연속 호출 테스트 (디바운스 트리거로 인해 씹혀야 정상)
    speak("전방에 의자가 있습니다.", 3, 2.0); 
    sleep(1);
    
    // 3. 비상 상황 모사 인터랩트 (말하던 의자 안내 끊고 즉시 출력)
    printf("\n⚠️ [위험 상황 발생 이벤트 캐치]\n");
    speak("긴급. 전방에 돌발 장애물이 있습니다.", 1, 0.0);
    
    // 자식 프로세스 처리 대기 후 종료
    sleep(4);
    printf("시스템 종료.\n");
    
    return 0;
}